torch/fx/experimental/symbolic_shapes.py

```diff
@@ -379,12 +407,17 @@ def rebind_unbacked(shape_env, n: torch.fx.Node, result):
             if (
                 isinstance(raw_u1, sympy.Piecewise)
                 and len(raw_u1.args) == 2
-                and raw_u1.args[0][0] == 1
-                and isinstance(eq := raw_u1.args[0][1], sympy.Eq)
+                and (
+                    raw_u1_args0 := cast(
```

Comment:
We need casting? We can't just specify the type here before the walrus op?

Reply:
Unfortunately, sympy annotates the type of args as Sequence[sympy.Basic], so the type annotation of args[0] is Basic when it needs to be a Tuple (because Piecewise). So even if we specified the type in walrus we would need to also do a mypy ignore.